rust/src/cliwrap/kernel_install.rs

```diff
@@ -64,13 +67,20 @@ fn redo_systemctl_wrap() -> Result<()> {
     Ok(())
 }
 
-#[context("Running dracut")]
-fn run_dracut(kernel_dir: &str) -> Result<()> {
+#[context("Generate initfs")]
+fn generate_initfs(kernel_dir: &str) -> Result<()> {
     let root_fs = Utf8Dir::open_ambient_dir("/", cap_std::ambient_authority())?;
     let tmp_dir = tempfile::tempdir()?;
     let tmp_initramfs_path = tmp_dir.path().join("initramfs.img");
 
-    let cliwrap_dracut = Utf8Path::new(cliwrap::CLIWRAP_DESTDIR).join("dracut");
+    let mut initfs_tool = String::new();
+    let initoverlayfs_path = Path::new(INITOVERLAY_INSTALL_CMD);
+    if initoverlayfs_path.exists() {
+        initfs_tool = INITOVERLAY_INSTALL_CMD.to_string()
+    } else {
+        initfs_tool = "dracut".to_string()
+    }
```

Comment:
This can be simplified to a one liner like this:

```suggestion
    let initfs_tool = Path::new(INITOVERLAY_INSTALL_CMD).exists().then_some(INITOVERLAY_INSTALL_COMMAND).unwrap_or("dracut");
```

It's definitely more idiomatic to assign the result of a condition; the use of `then_some` as a combinator is more a taste thing.

In fact an instance of this pattern is right below:

```
    let dracut_path = cliwrap_dracut
        .exists()
        .then(|| cliwrap_dracut)
        .unwrap_or_else(|| Utf8Path::new("dracut").to_owned());
```